wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/event/DestroyedMachineEventListener.java

```diff
@@ -30,37 +30,38 @@
 import static org.eclipse.che.api.machine.shared.dto.event.MachineStatusEvent.EventType.DESTROYED;
 
 /**
- * The class listens changing of machine status and perform some actions when status is changed.
+ * The class listens changing of machine status and perform some actions when existing machine destroyed.
  *
  * @author Dmitry Shnurenko
  */
 @Singleton
-public class MachineStateListener implements EventSubscriber<MachineStatusEvent> {
-    private static final Logger LOG = LoggerFactory.getLogger(MachineStateListener.class);
+public class DestroyedMachineEventListener implements EventSubscriber<MachineStatusEvent> {
+    private static final Logger LOG = LoggerFactory.getLogger(DestroyedMachineEventListener.class);
 
     private final WorkspaceManager workspaceManager;
     private final EventService     eventService;
 
     @Inject
-    public MachineStateListener(WorkspaceManager workspaceManager, EventService eventService) {
+    public DestroyedMachineEventListener(WorkspaceManager workspaceManager, EventService eventService) {
         this.workspaceManager = workspaceManager;
         this.eventService = eventService;
     }
 
     @Override
     public void onEvent(MachineStatusEvent event) {
-        String workspaceId = event.getWorkspaceId();
+        if (DESTROYED.equals(event.getEventType())) {
```

Comment:
I separated that checks here because this event listener should process only `DESTROYED` machine state. The fact of processing only dev-machine is internal logic of listener.
The same for `CreateMachineEventListener`